drivers/serial/uart_infineon_cat1.c

```diff
@@ -0,0 +1,467 @@
+
+	/* Store user callback info */
+	data->irq_cb = cb;
+	data->irq_cb_data = cb_data;
+
+	/* Register a uart general callback handler  */
+	cyhal_uart_register_callback(uart_obj, _uart_event_callback_irq_mode, (void *) dev);
+}
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+
+
+///////////////////////////////////////////////////////////////////////////////
+//                     UART API STRUCTURE                                    //
+///////////////////////////////////////////////////////////////////////////////
+static const struct uart_driver_api uart_cat1_driver_api = {
+	.poll_in = uart_cat1_poll_in,
+	.poll_out = uart_cat1_poll_out,
+	.err_check = uart_cat1_err_check,
+
+#ifdef CONFIG_UART_USE_RUNTIME_CONFIGURE
+	.configure = uart_cat1_configure,
+	.config_get = uart_cat1_config_get,
+#endif /* CONFIG_UART_USE_RUNTIME_CONFIGURE */
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	.fifo_fill = uart_cat1_fifo_fill,
+	.fifo_read = uart_cat1_fifo_read,
+	.irq_tx_enable = uart_cat1_irq_tx_enable,
+	.irq_tx_disable = uart_cat1_irq_tx_disable,
+	.irq_tx_ready = uart_cat1_irq_tx_ready,
+	.irq_rx_enable = uart_cat1_irq_rx_enable,
+	.irq_rx_disable = uart_cat1_irq_rx_disable,
+	.irq_tx_complete = uart_cat1_irq_tx_complete,
+	.irq_rx_ready = uart_cat1_irq_rx_ready,
+	.irq_err_enable = uart_cat1_irq_err_enable,
+	.irq_err_disable = uart_cat1_irq_err_disable,
+	.irq_is_pending = uart_cat1_irq_is_pending,
+	.irq_update = uart_cat1_irq_update,
+	.irq_callback_set = uart_cat1_irq_callback_set,
+#endif    /* CONFIG_UART_INTERRUPT_DRIVEN */
+
+#ifdef CONFIG_UART_ASYNC_API
+	.callback_set = uart_cat1_async_cb_set,
+	.tx = uart_cat1_async_tx,
+	.tx_abort = uart_cat1_async_tx_abort,
+	.rx_enable = uart_cat1_async_rx_enable,
+	.rx_disable = uart_cat1_async_rx_disable,
+	.rx_buf_rsp = uart_cat1_async_rx_buf_rsp,
+#endif  /* CONFIG_UART_ASYNC_API */
+};
+
+///////////////////////////////////////////////////////////////////////////////
+//                     UART DRIVER INIT MACROS                               //
+///////////////////////////////////////////////////////////////////////////////
+#define INFINEON_CAT1_UART_INIT(n)						  \
+	static struct uart_cat1_data INFINEON_CAT1_uart##n##_data = { 0 };	  \
+										  \
+	static int INFINEON_CAT1_uart##n##_init(const struct device *dev)	  \
+	{									  \
+		cy_rslt_t result;						  \
+		struct uart_config z_cfg;					  \
+										  \
+		z_cfg.baudrate = DT_INST_PROP(n, current_speed);		  \
+		z_cfg.parity = DT_ENUM_IDX(DT_DRV_INST(n), parity);		  \
+		z_cfg.stop_bits = DT_INST_PROP(n, stop_bits);			  \
+		z_cfg.data_bits = DT_INST_PROP(n, data_bits);			  \
+		z_cfg.flow_ctrl = DT_INST_PROP(n, hw_flow_control);		  \
+										  \
+		result = cyhal_uart_init(&DEV_DATA(dev)->obj,			  \
+					 (cyhal_gpio_t) DT_INST_PROP(n, tx_pin),  \
+					 (cyhal_gpio_t) DT_INST_PROP(n, rx_pin),  \
+					 (cyhal_gpio_t) DT_INST_PROP(n, cts_pin), \
+					 (cyhal_gpio_t) DT_INST_PROP(n, rts_pin), \
+					 NULL, NULL);				  \
```

Comment:
you'll need to implement a pinctrl driver, this is no longer allowed. please check https://github.com/zephyrproject-rtos/zephyr/issues/39740

Reply:
i will rework to use pinctrl data instead tx_pin, rx_pin, cts_pin, rts_pin.

Reply:
note: pinctrl data is opaque, so drivers are not allowed to use such information. `cyhal_uart_init` will likely need to be reworked to not initialize pins.

Reply:
do you mean, that i can not retrieve information about port and pin from pinctrl data ? or i can but it is not allowed...
I expected to create objects of cyhat_gpio_t  from pinctrl data, which can  be used for cyhal_uart_init().